packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -269,6 +305,46 @@ class VideoViewport extends Viewport implements IVideoViewport {
     return null;
   }
 
+  public setWindowLevel(windowWidth = 256, windowCenter = 128) {
+    this.windowLevel = { windowWidth, windowCenter };
+    this.setColorTransform();
+  }
+
+  public setAverageWhite(averageWhite: [number, number, number]) {
+    this.averageWhite = averageWhite;
+    this.setColorTransform();
+  }
+
+  public setColorTransform() {
+    if (!this.windowLevel && !this.averageWhite) {
+      this.feFilter = null;
+    } else {
+      const { windowWidth = 256, windowCenter = 128 } = this.windowLevel || {};
+      const white = this.averageWhite || [255, 255, 255];
+      const maxWhite = Math.max(...white);
+      const scaleWhite = white.map((c) => maxWhite / c);
+      // From the DICOM standard: ((x - (c - 0.5)) / (w-1) + 0.5) * (ymax- ymin) + ymin
+      // which is x/(w-1) - (c - 0.5) / (w-1) + 0.5  for this case
+      const wlScale = 255 / (windowWidth - 1);
+      const wlDelta = -(windowCenter - 0.5) / (windowWidth - 1) + 0.5;
+      console.log('scaleWhite=', scaleWhite, wlScale, wlDelta);
+      this.feFilter = `url('data:image/svg+xml,\
+      <svg xmlns="http://www.w3.org/2000/svg">\
+        <filter id="colour" color-interpolation-filters="linearRGB">\
+        <feColorMatrix type="matrix" \
+        values="\
+          ${scaleWhite[0] * wlScale} 0 0 0 ${wlDelta} \
+          0 ${scaleWhite[1] * wlScale} 0 0 ${wlDelta} \
+          0 0 ${scaleWhite[2] * wlScale} 0 ${wlDelta} \
+          0 0 0 1 0" />\
+        </filter>\
+      </svg>#colour')`;
```

Comment:
since you are doing this we need some sanitization happen for XSS

How about you always check those (width and center and average white etc.) are numbers and not texts

Reply:
They are all calculated values, you can't effectively XSS except if you call the entire set method on the filter/assign to the internal value, in whih case you are dead anyways.